### PR TITLE
WIP: Tendermint Debug Tool

### PR DESCRIPTION
**Pull request by alexanderbez**
closes: #3249

----

* [ ] Updated all relevant documentation in docs
* [ ] Updated all code comments where relevant
* [ ] Wrote tests
* [ ] Updated CHANGELOG_PENDING.md

### CODE REVIEW

**Comment by alexanderbez** [comment]
@ebuchman @melekes the `kill` command of this is essentially ready. Feel free to take an early look or even play with it. I still need to implement the `dump` command, add docs, and whatever unit tests I can.

**Comment by melekes** [comment]
Awesome 👍 I will review/try to play tomorrow.

**Comment by ebuchman** [comment]
@alexanderbez @melekes any update on this?

**Comment by marbar3778** [comment]
@alexanderbez any update on this?

File: Makefile [hunk 1]
```diff
@@ -113,6 +113,9 @@ get_deps_bin_size:
 	@find $(WORK) -type f -name "*.a" | xargs -I{} du -hxs "{}" | sort -rh | sed -e s:${WORK}/::g > deps_bin_size.log
 	@echo "Results can be found here: $(CURDIR)/deps_bin_size.log"
 
+tendermint-debug:
+	CGO_ENABLED=0 go install $(BUILD_FLAGS) -tags $(BUILD_TAGS) ./cmd/tendermintdb
```
[the remark this answers is not in the document]
Comment (marbar3778) [reply Makefile]: yes, this comes back around to https://github.com/tendermint/tendermint/issues/2495

[the remark this answers is not in the document]
Comment (thanethomson) [reply Makefile]: Commented on this in #2495 - would appreciate some thoughts on a good heuristic that we can agree on to decide whether something should be a subcommand or a separate tool altogether.

**Comment by melekes** [comment]
@alexanderbez do you have time to make the `tendermint-debug` a sub command of `tendermint` https://github.com/tendermint/tendermint/pull/3327#discussion_r281086321? Ideally we should either close this or merge.

**Comment by alexanderbez** [comment]
yes @melekes. I'll pick back up on this shortly.

**Pull request closed without merging**